### ISSUE TITLE
Pessimistic locking of project edits

### ISSUE THREAD

**Issue by elliottSlee** [bcgov/cas-cif#488]

**Comment by elliottSlee** [comment]
Following the ux research for #282, pessimistic locking is a good option to handle concurrent editing of project edits.

**Comment by pbastia** [comment]
Should we place the pessimistic lock on a single page / form, instead of the whole revision? Or would that cause issues if someone tries to submit something someone else is working on?

**Comment by LindsayMacfarlane** [comment]
Given the feedback from users and low probability of two people editing the same project, I'm wondering if this is more of a low priority that we can tackle later in development?

**Comment by pbastia** [comment]
Yes. However we'll still need to allow users to collaborate on revisions, which is just a small change: https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/bcgov/cas-cif/591

**Comment by elliottSlee** [comment]
> Yes. However we'll still need to allow users to collaborate on revisions, which is just a small change: https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/bcgov/cas-cif/591

Thumbs up buttercup!
